feature/move rr_viz to use custom_msgs package

rebeccaRossRobotics opened the pull request:
Tested on Envirobot, everything seems to be working. Needs testing on Dev platform also using the rossrobo pull request #46

Commit:
Move rr_viz to use rr_custom_msgs

Commit:
fixes

pepeRossRobotics commented:
I am happy with the changes :+1:

msadowski commented:
@JuliusSustarevas will need to correct me here if I'm wrong but I think this change will break it in a way. rr_viz was made open to the public due to qt licencing. If we remove the messages from here by putting them in ross_robo this will mean that it won't work for anyone else than us. This might still be OK from the licence perspective but it sounds like breaking the 'spirit' of open source  abit.

Thinking of our options here we can either:
1. Keep the relevant messages here - OK from the licence perspective, perhaps a bit annoying for us since we won't have all messages defined in one place
2. Make all our messages publicly available - I don't think there is any interesting IP there anyway but we would end up with yet another submodule
3. Split the messages into two packages, one public and one private - don't really like it, we will never know which package we should take the messages from.
4. Take the messages away and not care that it's public and noone can build it - probably not a nice image for us

pepeRossRobotics commented:
rr_custom_messages is a repository on its own, I think atm it is private, but the idea is for it to be public. the only issue that i can think of is if rr_viz has it as submodule or has an installation comment that the package is needed in the workspace

pepeRossRobotics commented:
so it is option 4 atm but after the initial PR will be option 2

JuliusSustarevas committed:
Added dependency in readme

Since you need rr_custom_msgs but it is not a submodule, I added this in readme

JuliusSustarevas commented:
Hi

There is another dependency issue here. 
Rviz_texturedSpehere is a submodule fo rr_perception inside the private ross_robo. 
I have searched for references to rviz_textured sphere and I believe **only** rrviz has this dependency since its the only visualisation tool software. 

I think lets add rviz_textured sphere as a submodule to this PR and  delete it from rr_perception.  
Pros- rrviz should build on its own (granted you have rr_custom_msgs which will **have to be public** at some point so that our customers can work with our ROS Network, which I believe is the intention eventually)  rr_viz building on its own is also in the spirit of making the "API" to the robot public and accessible. 

Cons- Creation of extra PRs. But since ross_robo subs rr_viz which would sub rviz_texture... then ross_robo should build and be usable on its own. So no real cons I think?

Commit:
add rviz_textured_sphere submodule

Commit:
Removed submodule rr_viz/submodules/rviz_textured_sphere

Commit:
Added submodule submodules/rviz_textured_sphere

rebeccaRossRobotics commented:
I've pushed moving rr_textured_sphere to be a submodule of rr_viz and there's an accompanying PR in ross_robo to remove it from rr_perception (no 53)

msadowski commented on rr_viz/CMakeLists.txt:
```diff
@@ -9,23 +9,12 @@ find_package(catkin REQUIRED COMPONENTS
-)
-
-generate_messages(DEPENDENCIES geometry_msgs std_msgs rr_viz)
+generate_messages(DEPENDENCIES geometry_msgs std_msgs)
```
generate_messages can probably be removed from here as well?

rebeccaRossRobotics replied:
done!

msadowski commented on rr_viz/README.md:
```diff
@@ -3,6 +3,9 @@
     The intention is to create widgets as needed and compose the GUI in a an easy way using QT designer/editor
 
+# Install/Dependencies
+- Must have rr_custom_msgs built and available
```
Since this package is open source is it time to open source rr_custom_msgs too?

rebeccaRossRobotics replied:
Yeah i think so, i pressume we need to give it a license in the package.xml though

msadowski commented on rr_viz/package.xml:
```diff
@@ -19,6 +19,7 @@
+  <build_depend>rr_custom_msgs</build_depend>
 
 
   <run_depend>actionlib</run_depend>
```
I can't comment at the exact line but "message_runtime" can probably be removed from here too.

rebeccaRossRobotics replied:
@msadowski  I just tried to build without the message_runtime line and it failed so will leave it in

msadowski replied:
this is very odd since it does not generate messages. I will create a jira issue for this so that we can check this later on.

Commit:
remove generate messages line
